packages/@mantine/modals/src/events.ts

```diff
@@ -40,44 +19,41 @@ export const openConfirmModal: ModalsEvents['openConfirmModal'] = (payload) => {
   return id;
 };
 
-export const openContextModal: ModalsEvents['openContextModal'] = <TKey extends MantineModal>(
-  payload: OpenContextModal<Parameters<MantineModals[TKey]>[0]['innerProps']> & {
-    modal: TKey;
-  } & DataAttributes
-) => {
+export const openContextModal: ModalsEvents['openContextModal'] = (payload) => {
   const id = payload.modalId || randomId();
   createEvent('openContextModal')({ ...payload, modalId: id });
   return id;
 };
 
-export const closeModal = createEvent('closeModal');
+export const closeModal: ModalsEvents['closeModal'] = createEvent('closeModal');
 
-export const closeContextModal: ModalsEvents['closeContextModal'] = <TKey extends MantineModal>(
-  id: TKey
-) => createEvent('closeContextModal')(id);
+export const closeContextModal: ModalsEvents['closeContextModal'] = (id) =>
+  createEvent('closeContextModal')(id);
 
-export const closeAllModals = createEvent('closeAllModals');
+export const closeAllModals: ModalsEvents['closeAllModals'] = createEvent('closeAllModals');
 
-export const updateModal = (payload: { modalId: string } & Partial<ModalSettings>) =>
+export const updateModal: ModalsEvents['updateModal'] = (payload) =>
   createEvent('updateModal')(payload);
 
-export const updateContextModal = (payload: { modalId: string } & Partial<OpenContextModal<any>>) =>
+export const updateContextModal: ModalsEvents['updateContextModal'] = (payload) =>
   createEvent('updateContextModal')(payload);
 
 export const modals: {
```

Comment:
This could also be unified to either remove or keep all mention of `Modal` at the end of each function. I left it like it because it would have required to modified some other files in demo, which was unecessary for the core of this PR. Also since every function is already exported, I'm not sure exporting `modals` is useful at all, as it could bring some confusion about the `modals` returned by `useModal`.